app/src/main/java/com/oatrice/internet_speed_testing/SpeedTestKotlin.kt

```diff
@@ -0,0 +1,115 @@
+package com.oatrice.internet_speed_testing
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.example.internet_speed_testing.InternetSpeedBuilder
+import com.example.internet_speed_testing.ProgressionModel
+import kotlinx.android.synthetic.main.layout_speedtest_ui.*
+import java.math.BigDecimal
+import java.text.DecimalFormat
+
+class SpeedTestKotlin : AppCompatActivity() {
+
+    var internetSpeedBuilder:InternetSpeedBuilder?=null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.layout_speedtest_ui)
+
+        internetSpeedBuilder = InternetSpeedBuilder(this)
+        setupView()
+    }
+
+    fun setupView(){
+        startButton.setOnClickListener {
+            buildTestSpeed()
+
+            if(startButton.text == "TEST AGAIN"){
```

Comment:
- This block I think you want to start testing again but in runtime is not test again. It still continues current testing"